pkg/clusterConfig/watchConfig.go

```diff
@@ -0,0 +1,61 @@
+package clusterConfig
+
+import (
+	policyv1 "github.com/liqoTech/liqo/api/cluster-config/v1"
+	"github.com/liqoTech/liqo/pkg/crdClient/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+	"os"
+	"path/filepath"
+)
+
+func WatchConfiguration(handler func(watch.Event), client *v1alpha1.CRDClient, kubeconfigPath string) {
+	if client == nil {
+		config, err := NewConfig(kubeconfigPath)
+		if err != nil {
+			klog.Error(err, err.Error())
+			os.Exit(1)
+		}
+
+		client, err = v1alpha1.NewFromConfig(config)
+		if err != nil {
+			klog.Error(err, err.Error())
+			os.Exit(1)
+		}
+	}
+
+	watcher, err := client.Resource("clusterconfigs").Watch(metav1.ListOptions{})
+	if err != nil {
+		klog.Error(err, err.Error())
+		os.Exit(1)
+	}
+
+	for event := range watcher.ResultChan() {
+		handler(event)
+	}
+}
+
+func NewConfig(kubeconfigPath string) (*rest.Config, error) {
```

Comment:
Why did you replicate this function from the client library? We have a shared library for this purpose